tests/validation/tests/v3_api/test_node_template.py

```diff
@@ -0,0 +1,283 @@
+import os
+import pytest
+from rancher import ApiError
+from .common import *  # NOQA
+from .test_rke_cluster_provisioning import create_and_validate_cluster
+import time
+
+DO_ACCESSKEY = os.environ.get('DO_ACCESSKEY', "None")
+engine_install_url = "https://releases.rancher.com/install-docker/19.03.sh"
+rke_config = {
+    "addonJobTimeout": 30,
+    "authentication":
+    {"strategy": "x509",
+     "type": "authnConfig"},
+    "ignoreDockerVersion": True,
+    "ingress":
+        {"provider": "nginx",
+         "type": "ingressConfig"},
+    "monitoring":
+        {"provider": "metrics-server",
+         "type": "monitoringConfig"},
+    "network":
+        {"plugin": "canal",
+         "type": "networkConfig",
+         "options": {"flannel_backend_type": "vxlan"}},
+    "services": {
+        "etcd": {
+            "extraArgs":
+                {"heartbeat-interval": 500,
+                 "election-timeout": 5000},
+            "snapshot": False,
+            "backupConfig":
+                {"intervalHours": 12, "retention": 6, "type": "backupConfig"},
+            "creation": "12h",
+            "retention": "72h",
+            "type": "etcdService"},
+        "kubeApi": {
+            "alwaysPullImages": False,
+            "podSecurityPolicy": False,
+            "serviceNodePortRange": "30000-32767",
+            "type": "kubeAPIService"}},
+    "sshAgentAuth": False}
+
+@if_test_rbac
+def test_rbac_node_template_create(remove_resource):
+    client = get_user_client()
+    node_template = node_template_do(client)
+    templates = client.list_node_template(name=node_template.name)
+    assert len(templates) == 1
+    time.sleep(.5)
+    remove_resource(node_template)
+
+
+@if_test_rbac
+def test_rbac_node_template_list(remove_resource):
```

Comment:
Can we also have a description for each testcase ?

Reply:
Each test case now has a small description